README.md

```diff
@@ -1,8 +1,114 @@
+token endpoint on the first retrieval. The cached token endpoint is not invalidated during operator runtime, but is updated when the IAS credentials or tenant URL are changed.
+
+### Referencing IAS applications by name
+The IAS application is created with a name that matches the name of the EventingAuth CR. This name is the unique runtime ID of the cluster for which the IAS application is created.
+Since we do not want to store the IAS application ID in the secret stored on the managed runtime , we can read the IAS application only by its name.  
+During the creation of the application, existing applications with the same name are read. If an application with the same name exists, it is deleted, as we assume this is due to a failed reconciliation.
+If more than one application with the same name already exists, the reconciliation fails. The same behaviour occurs when reconciling the deletion of the EventingAuth CR.
+
+It was decided not to delete any of the existing applications in this case, as it is an unexpected condition that may have been caused by manual actions, and we may want to keep the applications to find the cause of the issue.
+
+### Handling of failed IAS application and secret creation
+If the creation of the IAS application fails, the reconciliation will be retried. If an application has already been created, it is deleted before creation is attempted again.
+To avoid having multiple applications with the same name, the application is created again only if the deletion is successful.
+During the application creation process, there are several steps that can fail. First, the application is created, then the client secret is created, and finally the client ID of the client secret is read.    
+It was decided to always delete the application if any of these steps fail, as this makes the whole process more understandable and easier to maintain.  
+The reason for this is that the existing application can only be reused if the reconciliation failed before the client secret was successfully created, as we have no way to retrieve the client secret the next time the reconciliation is performed. 
+
+Additionally,if the creation of the secret on the managed runtime fails, we also delete the application and recreate it during the next reconciliation.
+In this case, there is still the option of not always deleting the application by caching the created application in the operator, but caching the application and thus the client secrets might not be a good idea either for security reasons.
+
+## Future Improvements
+- Identify IAS Application with its UUID. Currently, it is identified with its name, see [Referencing IAS applications by name](#referencing-ias-applications-by-name).
+Prevent IAS Application multiple recreation in case K8s secret creation fails, see [Handling of failed IAS application and secret creation](#handling-of-failed-ias-application-and-secret-creation).
```

Comment:
It'd be perfect if you could the following suggestion:
```suggestion
## Future Improvements
- Identify IAS Application with its UUID. Currently, it is identified with its name, see [Referencing IAS applications by name](#referencing-ias-applications-by-name).
- Prevent IAS Application multiple recreation in case K8s secret creation fails, see [Handling of failed IAS application and secret creation](#handling-of-failed-ias-application-and-secret-creation).
```